app/src/main/java/com/yacgroup/yacguide/TourbookActivity.kt

```diff
@@ -132,17 +118,70 @@ class TourbookActivity : BaseNavigationActivity() {
         }
     }
 
-    private fun _import() {
-        _ioOption = IOOption.eImport
-        if (FilesystemUtils.checkPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)) {
-            _showFileChooser()
+    @Throws(IOException::class)
+    private fun _readTextFromUri(uri: Uri): String {
+        val stringBuilder = StringBuilder()
```

Comment:
All this detailed stuff of file reading & writing does actually belong to the `TourbookExporter` class and not to this activity.